### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Aish94**
Query users by:
1. All users
2. By search keywords on first_name, last_name, email or username
3. Skills
Query Projects by:
1. All Projects
2. Title
3. Skills

### CODE REVIEW

File: schema/resolvers.js [hunk 1]
```diff
@@ -18,7 +18,42 @@ module.exports = {
       return getLoggedInUser(jwt_object);
     },
 
-    users: async (root, data, { models: { User } }) => User.findAll(data),
+    users: async (root, { search_string, skills }, { models: { User, Skill } }) => {
+      const query = {};
+      if (search_string) {
+        const search_params = [];
+        const search_keywords = search_string.split(' ');
```
Comment (the-vampiire) [inline schema/resolvers.js]: @Aish94 can you describe the `search_string` parameter and its use in the resolver?

File: schema/resolvers.js [hunk 2]
```diff
@@ -18,7 +18,42 @@ module.exports = {
...
+        const search_fields = ['first_name', 'last_name', 'username', 'email'];
+        for (let i = 0; i < search_keywords.length; i += 1) {
+          for (let j = 0; j < search_fields.length; j += 1) {
+            const search = {};
+            search[search_fields[j]] = { [Op.iLike]: `%${search_keywords[i]}%` };
+            search_params.push(search);
+          }
+        }
+        query.where = {
+          $or: [
```
Comment (TheSabby) [inline schema/resolvers.js]: No need for spread operator. It can just be `$or: search_params`

File: schema/resolvers.js [hunk 3]
```diff
@@ -18,7 +18,42 @@ module.exports = {
...
+        };
+      }
+      if (skills) {
+        query.include = [{
+          model: Skill,
+          where: {
+            name: skills,
+          },
+        },
+        ];
```
Comment (TheSabby) [inline schema/resolvers.js]: Fix indentation. Remove comma from line 46 and move line 47 up to line 46.

File: schema/resolvers.js [hunk 4]
```diff
@@ -18,7 +18,42 @@ module.exports = {
...
+      }
+
+      if (query) {
+        return User.findAll({
+          ...query });
```
Comment (TheSabby) [inline schema/resolvers.js]: No need for spread operator: `return User.findAll(query);`

File: schema/resolvers.js [hunk 5]
```diff
@@ -18,7 +18,42 @@ module.exports = {
...
+      }
+
+      return User.findAll();
```
Comment (TheSabby) [inline schema/resolvers.js]: Not sure but it is possible you don't need the previous if statement (`if (query) {...}`) and you can just use `return User.findAll(query);`

File: schema/resolvers.js [hunk 6]
```diff
@@ -55,7 +90,32 @@ module.exports = {
...
+          where: {
+            name: skills,
+          },
+        },
+        ];
+      }
+
+      if (query) {
+        return Project.findAll({
+          ...query });
```
Comment (TheSabby) [inline schema/resolvers.js]: Same comment as users resolver. No need for spread operator.

File: schema/resolvers.js [hunk 7]
```diff
@@ -55,7 +90,32 @@ module.exports = {
...
+      }
+
+      return Project.findAll();
```
Comment (TheSabby) [inline schema/resolvers.js]: Same comment as users resolver. No need for previous if statement.

**Pull request merged**